### ISSUE TITLE
Also complete session names in generated zsh completions

### ISSUE THREAD

**Issue by cyruseuros** [zellij-org/zellij#3083]
Exactly like in https://github.com/zellij-org/zellij/issues/1030 - personally I've never written any custom zsh completions so unfortunately I can't contribute here.

**Comment by sstark** [comment]
It is unfortunate that clap can not generate this. Also it does not seem to be straight forward to add it manually either. Instead, I am providing a patch file that you can download, which will add dynamic session completion to the attach/kill-session/delete-session commands. (https://gist.github.com/sstark/22318d37b8fc9a21c60a354eec460fff). It was generated using zellij 0.41.2.

First download the gist and save it to _zellij.patch. Then you can apply it to the generated completion. Here are the commands for zsh:

```
curl -O https://gist.githubusercontent.com/sstark/22318d37b8fc9a21c60a354eec460fff/raw/57f0d1d6bfe23e68813d2b8d96ba13c381c347bf/_zellij.patch
patch -s -o - =(zellij setup --generate-completion zsh) _zellij.patch > _zellij
```
Copy the resulting patched _zellij file to whereever you keep your completions.

**Comment by Lockszmith-GH** [comment]
Just wanted to say thanks to @sstark.

Below is a one-liner to generate the `_zellij` completion script:
```zsh
patch -s -o - \
    =(zellij setup --generate-completion zsh) \
    =(curl -sL https://gist.githubusercontent.com/sstark/22318d37b8fc9a21c60a354eec460fff/raw/57f0d1d6bfe23e68813d2b8d96ba13c381c347bf/_zellij.patch) \
    > ${MY_ZSH_COMLETION_PATH:-~/.local/share/zsh/completions}/_zellij
```

On one system, I actually do source it (because - reasons, not particularly good ones), and it looks like this:
```zsh
. <(
    patch -s -o - \
        =(zellij setup --generate-completion zsh) \
        =(curl -sL https://gist.githubusercontent.com/sstark/22318d37b8fc9a21c60a354eec460fff/raw/57f0d1d6bfe23e68813d2b8d96ba13c381c347bf/_zellij.patch) \
        | sed -Ee 's/^(_(zellij) ).*/compdef \1\2/'
)

#or

. <( sed -Ee 's/^(_(zellij) ).*/compdef \1\2/' ~/.local/share/lib/_zellij_zsh_lib )
```

Hope others find this helpful.

**Comment by Lockszmith-GH** [comment]
Zsh's latest version, since 2022, is 5.9 ([source](https://zsh.sourceforge.io/FAQ/zshfaq01.html#l6))

I'm using these zsh versions flawlessly:
```
$ zsh --version # output from different machines
 
zsh 5.9 (arm64-apple-darwin24.0)
zsh 5.8 (x86_64-redhat-linux-gnu)
zsh 5.9 (x86_64-debian-linux-gnu)
```

@arrrgi what version are you using?

**Comment by arrrgi** [comment]
It does patch, but it also has a mismatch leading to the following on the last line of the file

```
}1 out of 5 hunks failed--saving rejects to -.rej
```

**Comment by Lockszmith-GH** [comment]
you are correct, there is an error, however, it only fails to patch the zpipe command, not a critical issue.
the session list is still loaded.

I'll try and create a proper correction later, in the meantime you can use 2> /dev/null to suppress the error message.

**Comment by sstark** [comment]
If you look at the patch, it's fairly easy to manually add the changes to the output of zellij's generated completion. I am not planning to update this, as I am not using zellij currently, sorry. Also, this should be fixed in zellij properly.
